SofaKernel/SofaFramework/src/sofa/config.h.in

```diff
@@ -113,6 +113,22 @@ typedef unsigned __int64	uint64_t;
 #define sofa_do_tostring(a) #a
 #define sofa_tostring(a) sofa_do_tostring(a)
 
+///////////////////// These two macro can be used to convert deprecation attributes as error.
+#if defined(_MSC_VER)
+#define BEGIN_DEPRECATION_AS_ERROR _Pragma("warning( push )") \
```

Comment:
Should we prefix macro with SOFA?

BEGIN_DEPRECATION_AS_ERROR -> SOFA_BEGIN_DEPRECATION_AS_ERROR
END_DEPRECATION_AS_ERROR -> SOFA_END_DEPRECATION_AS_ERROR